### PR TITLE
Ab#1738 api view users

### PR DESCRIPTION
**Pull request by tschia**

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/hlth/mohums/controller/GroupController.java [hunk 1]
```diff
@@ -14,8 +14,10 @@ public GroupController(WebClientService webClientService) {
         this.webClientService = webClientService;
     }
 
+    private final String groupsPath = "/groups";
+
     @GetMapping("/groups")
     public Mono<Object> groups() {
-        return webClientService.getGroups();
+        return webClientService.get(groupsPath, null);
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/mohums/controller/GroupController.java:21]: refactored in webclient service to use a single method

File: backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/WebClientConfig.java [hunk 2]
```diff
@@ -64,8 +64,8 @@ public WebClient webClient(ReactiveOAuth2AuthorizedClientManager authorizedClien
      */
     private ExchangeFilterFunction logRequest() {
         return ExchangeFilterFunction.ofRequestProcessor(c -> {
-            webClientLogger.info("Request: {} {}", c.method(), c.url());
-            c.headers().forEach((n, v) -> webClientLogger.info("request header {}={}", n, v));
+            webClientLogger.debug("Request: {} {}", c.method(), c.url());
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/WebClientConfig.java:67]: don't print the token on every request

File: backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/WebClientConfig.java [hunk 3]
```diff
@@ -64,8 +64,8 @@ public WebClient webClient(ReactiveOAuth2AuthorizedClientManager authorizedClien
...
+            c.headers().forEach((n, v) -> webClientLogger.debug("request header {}={}", n, v));
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/WebClientConfig.java:68]: don't print the token on every request

File: backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/WebClientService.java [hunk 4]
```diff
@@ -26,10 +28,13 @@ public WebClientService(WebClient kcAuthorizedWebClient) {
                 .flatMapMany(r -> r.bodyToFlux(ClientRepresentation.class));
     }
 
-    public Mono<Object> getGroups() {
+    public Mono<Object> get(String path, MultiValueMap<String, String> queryParams) {
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/WebClientService.java:31]: refactored to be single reusable method

File: backend/src/main/resources/application.yaml [hunk 5]
```diff
@@ -16,7 +17,7 @@ spring:
         registration:
           keycloak:
             client-id: user-management-service
-            client-secret: 
+            client-secret: ${UMS_CLIENT_SECRET}
```
Comment (tschia) [inline backend/src/main/resources/application.yaml:20]: now uses environment variable

File: backend/src/test/java/ca/bc/gov/hlth/mohums/integration/MoHUmsIntegrationTests.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureWebTestClient
+public class MoHUmsIntegrationTests {
```
Comment (tschia) [inline backend/src/test/java/ca/bc/gov/hlth/mohums/integration/MoHUmsIntegrationTests.java:27]: integration tests check:
 - no unauthorized access to secure endpoints
 - authorized access works (get a 200)
 - unsecure endpoint does not require a jwt

File: backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import ca.bc.gov.hlth.mohums.webclient.WebClientService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+
+@RestController
+public class UsersController {
```
Comment (ahoplock) [inline backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java:14]: We named the other controllers singular (e.g. GroupController, ClientController). The naming convention should be the same, not sure which way it should go though...

Comment (tschia) [reply backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java:14]: renamed to GroupsController and ClientsController

**Review by ahoplock — COMMENTED** [review]
Reviewed and approved.

**Pull request merged**